homeassistant/components/seventeentrack/sensor.py

```diff
@@ -1,341 +1,160 @@
 
-    try:
-        login_result = await client.profile.login(
-            config[CONF_USERNAME], config[CONF_PASSWORD]
-        )
+    async_add_entities(entities)
 
-        if not login_result:
-            _LOGGER.error("Invalid username and password provided")
-            return
-    except SeventeenTrackError as err:
-        _LOGGER.error("There was an error while logging in: %s", err)
-        return
 
-    scan_interval = config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+class SeventeenTrackBaseSensor(CoordinatorEntity, SensorEntity):
+    """Base class for SeventeenTrack sensors."""
 
-    data = SeventeenTrackData(
-        client,
-        async_add_entities,
-        scan_interval,
-        config[CONF_SHOW_ARCHIVED],
-        config[CONF_SHOW_DELIVERED],
-        str(hass.config.time_zone),
-    )
-    await data.async_update()
-
-
-class SeventeenTrackSummarySensor(SensorEntity):
-    """Define a summary sensor."""
-
-    _attr_icon = "mdi:package"
+    coordinator: SeventeenTrackDataCoordinator
+    _attr_icon = ICON
     _attr_native_unit_of_measurement = "packages"
 
-    def __init__(self, data, status, initial_state):
-        """Initialize."""
-        self._attr_extra_state_attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        self._data = data
-        self._state = initial_state
-        self._status = status
-        self._attr_name = f"Seventeentrack Packages {status}"
-        self._attr_unique_id = f"summary_{data.account_id}_{slugify(status)}"
+    def __init__(self, coordinator: SeventeenTrackDataCoordinator) -> None:
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self._attr_name: str = coordinator.config_entry.data[CONF_NAME]
+        self._account_id: str = coordinator.account_id
+        self._attrs = {ATTR_ATTRIBUTION: ATTRIBUTION}
 
     @property
-    def available(self):
-        """Return whether the entity is available."""
-        return self._state is not None
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this 17Track instance."""
+        return {
+            "identifiers": {(DOMAIN, self._account_id)},
+            "default_name": self.coordinator.config_entry.data[CONF_NAME],
+            "manufacturer": "17Track",
+            "entry_type": DeviceEntryType.SERVICE,
+        }
 
-    @property
-    def native_value(self):
-        """Return the state."""
-        return self._state
 
-    async def async_update(self):
-        """Update the sensor."""
-        await self._data.async_update()
+class SeventeenTrackPackagesSensor(SeventeenTrackBaseSensor):
+    """Define sensor for all packages."""
+
+    def __init__(self, coordinator: SeventeenTrackDataCoordinator) -> None:
+        """Initialize package sensor."""
+        super().__init__(coordinator)
+        self._attr_name = f"{self._attr_name} all packages"
+        self._attr_unique_id = f"{self._account_id}-{slugify(self._attr_name)}"
```

Comment:
Don't include the entity name in the unique_id. Just suffix the entity kind to the account id.